cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -443,6 +444,13 @@ func main() {
 		defer fuse.Close()
 	} else {
 		updates := make(chan string)
+		go func() {
+			c := make(chan os.Signal, 1)
+			signal.Notify(c, os.Interrupt)
+			_ = <-c // block
+			logging.Infof("Received SIGINT. Shutting down listener")
+			close(updates)
```

Comment:
The doc comment on Client.Run() suggests that only connSrc needs to be closed to get it to exit; if you leave the `updates` channel open, will it still work? (see my suggestion below)

Reply:
I can't close connSrc because it is a receive-only channel